### PR TITLE
feat: add chips for search

### LINKED ISSUE #1517: Add Horizontal Chips
**Describe the feature you'd like**
Add horizontal scrolling chips similar to Eventbrite
**Screenshots**
screenshot-1554257185656

**Would you like to work on the issue?**
- [x] Yes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #1517 

Changes: Added Chips in searchResultFragment

Screenshots for the change:
videotogif_2019 04 03_10 10 07

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal please review

File: app/schemas/org.fossasia.openevent.general.OpenEventDatabase/1.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "formatVersion": 1,
   "database": {
     "version": 1,
-    "identityHash": "0b3cd25764884626e03f56b0600d1c76",
+    "identityHash": "2eccdc20f2068430bba84bf884a2ec02",
```
Comment (liveHarshit) [inline app/schemas/org.fossasia.openevent.general.OpenEventDatabase/1.json]: Do not commit this file.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt [hunk 2]
```diff
@@ -11,23 +11,29 @@ import androidx.core.view.isVisible
...
+import org.fossasia.openevent.general.common.EventClickListener
+import org.fossasia.openevent.general.common.FavoriteFabClickListener
+import org.fossasia.openevent.general.common.ShareFabClickListener
 import org.fossasia.openevent.general.di.Scopes
 import org.fossasia.openevent.general.event.Event
-import org.fossasia.openevent.general.common.EventClickListener
 import org.fossasia.openevent.general.event.EventDetailsFragmentArgs
 import org.fossasia.openevent.general.event.EventUtils
-import org.fossasia.openevent.general.common.FavoriteFabClickListener
-import org.fossasia.openevent.general.common.ShareFabClickListener
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt]: Unnecessary changes

File: app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt [hunk 3]
```diff
@@ -53,7 +59,38 @@ class SearchResultsFragment : Fragment() {
...
+            }
+            getString(R.string.month) -> rootView.month_chip.apply {
+                isChecked = true
+                chipBackgroundColor = resources.getColorStateList(R.color.colorPrimary)
+            }
+        }
 
+        rootView.chipGroup.setOnCheckedChangeListener { chipGroup, id ->
+            for (i in 0 until chipGroup.childCount) {
+                val chip = chipGroup.getChildAt(i) as Chip
```
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt]: You can use smart cast instead.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt [hunk 4]
```diff
@@ -68,7 +67,7 @@ class SearchViewModel(
                 |       'name':'location-name',
                 |       'op':'ilike',
                 |       'val':'%$location%'
-                |    }, {
+                |    }, {./
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt]: Why these changes?

File: app/src/main/res/layout/fragment_search_results.xml [hunk 5]
```diff
@@ -65,6 +65,49 @@
 
             <include layout="@layout/content_no_internet" />
 
+            <HorizontalScrollView
+                android:layout_width="wrap_content"
+                android:scrollbars="none"
+                android:paddingLeft="4dp"
+                android:layout_height="wrap_content"
+                android:paddingStart="4dp">
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_search_results.xml]: Dimensions from dimension resoures file.

**Comment by aggarwalpulkit596** [comment]
@liveHarshit please review

File: app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt]: Same.

[the remark this answers is not in the document]
Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchResultsFragment.kt]: Smart cast can't work here as chipbackgroundcolor is only the property of chip view and it the compiler doesn't understand it untill explicit cast is done

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal anything left out ?
